qiskit_addon_obp/backpropagation.py

```diff
@@ -164,7 +164,8 @@ def handle_timeout(signum, frame):
             # PERF: we will likely need to parallelize this loop
             for i in range(num_observables):
                 non_trivial_slice = False
-                for op_idx, op_node in enumerate(circuit_to_dag(slice_).topological_op_nodes()):
+                op_nodes = reversed(list(circuit_to_dag(slice_).topological_op_nodes()))
```

Comment:
Must we explicitly convert the iterator to a list? Or can we do the following instead?
```suggestion
                op_nodes = reversed(circuit_to_dag(slice_).topological_op_nodes())
```